sisl/geom/nanoribbon.py

```diff
@@ -154,3 +159,461 @@ def zgnr(width, bond=1.42, atoms=None):
     agnr : armchair graphene nanoribbon
     """
     return graphene_nanoribbon(width, bond, atoms, kind='zigzag')
+
+
+Heteroribbon_section = namedtuple("Section", ["W", "L", "shift", "atoms"], defaults=(None, 1, 0, None))
```

Comment:
I would prefer this being a local variable. In that way there is no confusion, and defaults can be changed for future stuff, this would force them to use the function attribute.

Also, the `defaults=(1, 0, None)` suffices, that would mean `W` is required argument.

Reply:
I would just prefix the variable as `_heteroribbon_section` to ensure it isn't *easily* imported/known.